File: generated-code/csharp/codecraft/TcpReadWrite/Model/RepairAction.cs
namespace TransGenTest.Model
{
    /// <summary>
    /// Repair action
    /// </summary>
    public struct RepairAction
    {
        /// <summary>
        /// Target entity's ID
        /// </summary>
        public int Target { get; set; }
    
        public RepairAction(int target)
        {
            this.Target = target;
        }
    
        /// <summary> Read RepairAction from reader </summary>
        public static RepairAction ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new RepairAction();
            result.Target = reader.ReadInt32();
            return result;
        }
    
        /// <summary> Write RepairAction to writer </summary>
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(Target);
        }
    
        /// <summary> Get string representation of RepairAction </summary>
        public override string ToString() {
            string stringResult = "RepairAction { ";
            stringResult += "Target: ";
            stringResult += Target.ToString();
            stringResult += " }";
            return stringResult;
        }
    }
}

File: generated-code/rust/codecraft/TcpReadWrite/src/model/repair_action.rs
use super::*;

/// Repair action
#[derive(Clone, Debug, PartialEq, Eq, Hash)]
pub struct RepairAction {
    /// Target entity's ID
    pub target: i32,
}

impl trans::Trans for RepairAction {
    fn write_to(&self, writer: &mut dyn std::io::Write) -> std::io::Result<()> {
        self.target.write_to(writer)?;
        Ok(())
    }
    fn read_from(reader: &mut dyn std::io::Read) -> std::io::Result<Self> {
        let target: i32 = trans::Trans::read_from(reader)?;
        Ok(Self {
            target,
        })
    }
}